desktop/core/src/desktop/lib/botserver/views_tests.py

```diff
@@ -32,36 +33,17 @@
 
 LOG = logging.getLogger(__name__)
 
-if conf.SLACK.IS_ENABLED.get():
-  class TestBotServer(unittest.TestCase):
-    def test_get_bot_id(self):
-      with patch('desktop.lib.botserver.views.slack_client.api_call') as api_call:
-        api_call.return_value = {
-          'members': [
-            {
-              'name': 'hue_bot',
-              'deleted': False,
-              'id': 'U01K99VEDR9'
-            }
-          ]
-        }
-        assert_equal(get_bot_id('hue_bot'), 'U01K99VEDR9')
+class TestBotServer(unittest.TestCase):
+  
+  @classmethod
+  def setUpClass(cls):
+    if not conf.SLACK.IS_ENABLED.get():
+      raise SkipTest
 
-        api_call.return_value = {
-          'members': [
-            {
-              'name': 'hue_bot',
-              'deleted': True,
-              'id': 'U01K99VEDR9'
-            }
-          ]
-        }
-        assert_equal(get_bot_id('hue_bot'), None)
-
-    def test_say_hi_user(self):
-      with patch('desktop.lib.botserver.views.slack_client.api_call') as api_call:
-        api_call.return_value = {
-          "ok": True
-        }
-        response = say_hi_user("channel", "user_id")
-        assert_true(response['ok'])
\ No newline at end of file
+  def test_say_hi_user(self):
+    with patch('desktop.lib.botserver.views.slack_client.api_call') as api_call:
+      api_call.return_value = {
+        "ok": True
+      }
+      response = say_hi_user("channel", "user_id")
+      assert_true(response['ok'])
```

Comment:
Why don't we test the bot_message by the way?

Reply:
We were returning the response from the api call to the parse_events().
Would it be nice if we return the botmessage instead and do the api call in parse_events() itself?

That way we can test the "botmessage" and also the error if api call raises( which I was thinking of making a UT for parse_events() too ).

Is this refactoring good @romainr ?

Reply:
It is good, just FYI I was saying that you could check that we send the proper message:

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with

(this is obviously more for learning, will be useful later for more critical stuff ;)

Reply:
Added the test for checking the message.
Thank you for this learning! 😄

Reply:
:+1: